contracts/core/libraries/Clone.sol

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.19;
+
+import "@openzeppelin-upgradeable/proxy/ClonesUpgradeable.sol";
+
+/// @title Clone library
+/// @notice A library to create clones of the strategy contracts when a pool is created
+library Clone {
+    /// @notice Checks if the address is a contract
+    /// @param _address The address to check
+    function _isContract(address _address) internal view returns (bool) {
+        uint256 codeSize;
+        assembly {
+            codeSize := extcodesize(_address)
+        }
+        return codeSize > 0;
+    }
+
+    /// @notice Create a determenstic clone of of contract
+    /// @param _contract The address of the contract to clone
+    /// @param _nonce The nonce to use for the clone
+    function createClone(address _contract, uint256 _nonce) internal returns (address clone) {
+        require(_isContract(_contract), "not a contract");
+        bytes32 salt = keccak256(abi.encodePacked(msg.sender, _nonce++));
+        clone = ClonesUpgradeable.cloneDeterministic(_contract, salt);
```

Comment:
What's the reasoning to use `cloneDeterministic` instead of `clone`? 

That would save needing a nonce at all, and not sure what they goal is because you can't actually trust the determinism, since another use could increment the nonce before you.

If determinism is important, than you'll want a separate nonce per users to make sure it's reliable.

Reply:
Is there a reason we need to determine the address for the strategies @thelostone-mc @KurtMerbeth

Reply:
Nah we could just use clones.

Reply:
fixed via ce70255

Reply:
@KurtMerbeth That Sherlock issue is mostly factually incorrect. If there was a reorg, then the transaction to send funds would be wound back as well.

However, I do see the point where there might be some risk of it being funded separate from the initial create but within 15-30 seconds, lining up with a reorg, having the funding txs be replayable, and then having an attacker create a new pool with the old address and running the replayable funding tx.

Might be safer to go back to the old way. Seems almost impossible to happen but cost of tracking the nonces is pretty small, so maybe can't hurt.

Reply:
Hmm alright. Given that we wanted to go the deterministic way :
7d5ce5b